src/Groups/GAPGroups.jl

```diff
@@ -2434,6 +2434,44 @@ function (G::FPGroup)(pairs::AbstractVector{Pair{T, S}}) where {T <: IntegerUnio
    return FPGroupElem(G, w)
 end
 
+"""
+    (G::FPGroup)(letters::AbstractVector{<:Integer})
+
+Return the element `x` of the full finitely presented group `G`
+that is described by `letters` as a list of integers, each entry corresponding to
+a group generator. Inverses of the generators are represented by negative
+numbers, see [`letters`](@ref).
+
+# Examples
+```jldoctest
+julia> G = free_group(2); lett = [1, 1, 1, -2];
+
+julia> x = G(lett)
+f1^3*f2^-1
+
+julia> letters(x) == lett
+true
+```
+"""
+function (G::FPGroup)(letters::AbstractVector{<:IntegerUnion})
+  n = ngens(G)
+  for l in unique(letters)
+    @req 0 < abs(l) <= n "generator number is at most $n"
+  end
```

Comment:
```suggestion
  @req all(l -> 0 < abs(l) <= n, letters) "invalid generator index"
```
the `unique` is probably more expensive than repeated checks. `all` makes this a bit shorter. `@req` error messages a slightly more efficient if they don't reference any local variable, but are instead a constant string. IMO the `n` does not have much benefit for a user, so I suggest to remove it.

Reply:
I agree, i've changed this now, thanks!